HerPublicWebsite.BusinessLogic/QuestionnaireUpdater.cs

```diff
@@ -105,10 +105,17 @@ public async Task<Questionnaire> GenerateReferralAsync(Questionnaire questionnai
 
         var referralRequest = new ReferralRequest(questionnaire);
         referralRequest = await dataAccessProvider.PersistNewReferralRequestAsync(referralRequest);
-        
+
         questionnaire.Hug2ReferralId = referralRequest.ReferralCode;
         questionnaire.ReferralCreated = referralRequest.RequestDate;
 
         return questionnaire;
     }
+
+    public async Task<Questionnaire> AddFutureNotificationContactDetailsAsync(Questionnaire questionnaire, string emailAddress)

```

Comment:
With the changes for 554 this could be an overload of `RecordNotificationConsentAsync`